wgpu-hal/src/dx12/command.rs

```diff
@@ -1272,13 +1273,185 @@ impl crate::CommandEncoder for super::CommandEncoder {
+                                .resource
+                                .GetGPUVirtualAddress()
+                                + instances.offset as u64
+                        },
+                    },
+                    instances.count,
+                ),
+                AccelerationStructureEntries::Triangles(triangles) => {
+                    geometry_desc = Vec::with_capacity(triangles.len());
+                    for triangle in triangles {
+                        geometry_desc.push(Direct3D12::D3D12_RAYTRACING_GEOMETRY_DESC {
+                            Type: Direct3D12::D3D12_RAYTRACING_GEOMETRY_TYPE_TRIANGLES,
+                            Flags: conv::map_acceleration_structure_geometry_flags(triangle.flags),
+                            Anonymous: Direct3D12::D3D12_RAYTRACING_GEOMETRY_DESC_0 {
+                                Triangles: Direct3D12::D3D12_RAYTRACING_GEOMETRY_TRIANGLES_DESC {
+                                    Transform3x4: triangle.transform.as_ref().map_or(
+                                        0,
+                                        |transform| unsafe {
+                                            transform.buffer.resource.GetGPUVirtualAddress()
+                                                + transform.offset as u64
+                                        },
+                                    ),
+                                    IndexFormat: triangle
+                                        .indices
+                                        .as_ref()
+                                        .map_or(Dxgi::Common::DXGI_FORMAT_UNKNOWN, |indices| {
+                                            conv::map_index_format(indices.format)
+                                        }),
+                                    VertexFormat: conv::map_acceleration_structure_vertex_format(
+                                        triangle.vertex_format,
+                                    ),
+                                    IndexCount: triangle
+                                        .indices
+                                        .as_ref()
+                                        .map_or(0, |indices| indices.count),
+                                    VertexCount: triangle.vertex_count,
+                                    IndexBuffer: triangle.indices.as_ref().map_or(
+                                        0,
+                                        |indices| unsafe {
+                                            indices
+                                                .buffer
+                                                .expect("needs buffer to build")
+                                                .resource
+                                                .GetGPUVirtualAddress()
+                                                + indices.offset as u64
+                                        },
+                                    ),
```

Comment:
Can we pull all of these complex assignments out of the struct init and make them local variables in the for loop - we can then have a place for computation and a place for initializing the struct. Should help reduce rightward creep too.

Reply:
This applies broadly for other places in the PR too - give a pass over and see if right shift can be reduced.

Reply:
I think I've addressed this here (and also in the similar version in `get_acceleration_structure_build_sizes`), but you may have a different opinion. I'll keep this unresolved.